crates/ra_hir/src/nameres/tests.rs

```diff
@@ -19,6 +19,19 @@ fn item_map(fixture: &str) -> (Arc<ItemMap>, ModuleId) {
     (db.item_map(krate.crate_id), module_id)
 }
 
+fn item_map_custom_crate_root(fixture: &str, root: &str) -> (Arc<ItemMap>, ModuleId) {
+    let (mut db, pos) = MockDatabase::with_position(fixture);
+
+    let mut crate_graph = CrateGraph::default();
+    crate_graph.add_crate_root(db.file_id(root));
```

Comment:
I think this is the same as `pos.file_id`? That is, you don't need to add `fn file_id` to MockDB. In fact, `FileMap` is a dead code, I should remove it :)

Reply:
Ah, that makes it easier, the FileMap was added by me. MockDatabase just used a incremental counter.

Reply:
On second thought that would require you to leave the position marker on the crate root I think.

Reply:
but that what happens anyway?

https://github.com/rust-analyzer/rust-analyzer/pull/690/files#diff-2e7269539c8928d62c4fb71200a3a2a9R158

Reply:
That is, true. I suppose I might be a conflict if someone wants to use `item_map_custom_crate_root` later. If you prefer to use the marker position I can document that in the function though.

Reply:
Yeah, let's just use a marker position.

In my experience, such "test setup functions" always feel a little bit ad-hock and non-orthogonal, but it do not cause much problems in practice. 

If we'd like, we can get the current behavior of mathcing by file name by iterating through source roots, though I'd reuse existing cursor marker infra